packages/flutter_tools/lib/runner.dart

```diff
@@ -88,6 +88,24 @@ Future<int> run(
           await globals.analytics.setTelemetry(value);
         }
 
+        // Enable analytics if user passes in the `--enable-telemetry` option
+        // `flutter --enable-telemetry`
+        //
+        // Same functionality as `flutter config --analytics` for enabling
+        // except with the `value` hard coded as true
+        if (args.contains('--enable-telemetry')) {
+          const bool value = true;
```

Comment:
I don't think this needs to exist: you can just pass true to the calls below.

Reply:
I was planning to leave this essentially identical to what we have in `packages/flutter_tools/lib/src/commands/config.dart` for disabling legacy analytics for now and then simplify the code once we have completely migrated away from the legacy analytics system. I left a TODO in there for me to make the change later. Thoughts on keeping this in until then?

Code below from config.dart

https://github.com/eliasyishak/flutter/blob/124411-disable-telemetry-refinements/packages/flutter_tools/lib/src/commands/config.dart#L129-L147

Reply:
I'm walking that back actually, it's still clear without hard coding a `value` variable, remove now

Reply:
I see what you're saying, but I don't think keeping it in is adding anything. It just makes the code in this conditional more complex and harder to read.